packages/batch-nft-reveal/contracts/NFTCollection.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
+import "@chainlink/contracts/src/v0.8/KeeperCompatible.sol";
+import "base64-sol/base64.sol";
+import "./interfaces/IERC20.sol";
+
+contract NFTCollection is
+    Ownable,
+    ERC721,
+    VRFConsumerBase,
+    KeeperCompatibleInterface
+{
+    // STRUCTS
+
+    struct Metadata {
+        uint256 startIndex;
+        uint256 endIndex;
+        uint256 entropy;
+    }
+
+    // IMMUTABLE STORAGE
+
+    uint256 public immutable maxSupply;
+    uint256 public immutable mintCost;
+    address public immutable linkToken;
+    uint256 internal immutable linkFee;
+    bytes32 internal immutable keyHash;
+
+    // MUTABLE STORAGE
+
+    uint256 public totalSupply = 0;
+    uint256 public revealedCount = 0;
+    uint256 public revealBatchSize;
+    uint256 public revealInterval;
```

Comment:
Can we have some comments that say that this interval is measure in seconds similar to the one in our NFTCollection repo where we specify that percents have 2 additional zeroes?